train/train.py

```diff
@@ -1,39 +1,146 @@
+import numpy as np
+import pandas as pd
+import pickle as pkl
+from pathlib import Path
+from collections import defaultdict
+
+from models import get_model_class
 
 
 def train(
         sites,
         models,
-        predictors
+        predictors=None,
+        predictors_paths=None,
+        log_metrics=["pr2", "nmae"],
+        inner_cv=5,
+        n_iter=20,
+        overwrite_existing_models=False
 ):
     """
     Train models using predictors on sitees.
 
     Args:
-        sites (str): Comma-separated list of site IDs to train on.
-                     Must match the name(s) of the data directories.
-        models (str): Comma-separated list of model names to train.
-                      Options: ['rf', 'ann', 'lasso', 'xgb']
-        predictors (str): Either a comma-separated list of predictors
-                          or a path to a file with predictor names.
-                          See train/predictors.json for an example.
+        sites (list<str>): Comma-separated list of site IDs to train on.
+                           Must match the name(s) of the data directories.
+        models (list<str>): Comma-separated list of model names to train.
+                            Options: ['rf', 'ann', 'lasso', 'xgb']
+        predictors (list<str>): Comma-separated list of predictors. Ignored if
+                                predictors_path is provided.
+        predictors_paths (list<str>): Comma-separated list path file(s) with
+                                      predictor names. See train/predictors.txt
+                                      for an example.
+        log_metrics (list<str>): Validation metrics to log.
+        inner_cv (int): Number of folds for k-fold cross validation in the
+                        training set(s) for selecting model hyperparameters.
+        n_iter (int): Number of parameter settings that are sampled in the
+                      inner cross validation.
+        overwrite_existing_models (bool): Whether to overwrite models if they
+                                          already exist.
 
     Saves trained models to data/{SiteID}/{model}/{predictor}/
     for each {SiteID}, {model}, and {predictor} subset.
     """
     data_dir = Path("data/")
-    sites = sites.split(",")
-    models = models.split(",")
+    if isinstance(sites, str):
+        sites = sites.split(",")
+    if isinstance(models, str):
+        models = models.split(",")
```

Comment:
Need to check if the input model is included and raise ValueError early, otherwise, an Error will pop out in Line 114.

Reply:
Good idea, done!